### PR TITLE
[BUGFIX] Default selection for variables of type list

### LINKED ISSUE #1312: Default selection for variables of type list
If the default_value field of a variables is not set, the default selection is equal to "", even if the list contains some values. 
I would naturally expect to see the first value as a default selection.  

If the All selection is allowed, we could also set All as default selection, but that's a bit more opinionated.

### PR DESCRIPTION
**Pull request by celian-garcia**
Encapsulates the complex part which manages the default selected value of a list variable into a function. 
It allowed me to make some interesting tests and fixes.
 
fixes #1312 

It's implementing the flow described in the issue + an additional behavior : when the default value or value from query param is not present in the list, it falls back to the first value of the list. 

# Checklist

- [x] Pull request has a descriptive title and context useful to a reviewer.
- [x] Pull request title follows the `[<catalog_entry>] <commit message>` naming convention using one of the following `catalog_entry` values: `FEATURE`, `ENHANCEMENT`, `BUGFIX`, `BREAKINGCHANGE`, `IGNORE`.
- [x] All commits have [DCO signoffs](https://github.com/probot/dco#how-it-works).

## UI Changes

- [ ] ~~Changes that impact the UI include screenshots and/or screencasts of the relevant changes.~~
> No screenshots as it is quite hard to capture in a screen. I let people reviewing the PR play with it locally.
- [x] Code follows the [UI guidelines](https://github.com/perses/perses/blob/main/ui/ui-guidelines.md).
- [x] Visual tests are stable and unlikely to be flaky. See [Storybook](https://github.com/perses/perses/tree/main/ui/storybook#visual-tests) and [e2e](https://github.com/perses/perses/tree/main/ui/e2e#visual-tests) docs for more details. Common issues include:
  - Is the data inconsistent? You need to mock API requests.
  - Does the time change? You need to use consistent time values or mock time utilities.
  - Does it have loading states? You need to wait for loading to complete.

### CODE REVIEW

File: ui/dashboards/src/components/Variables/TemplateVariable.tsx [hunk 1]
```diff
@@ -74,29 +78,76 @@ function ListVariable({ name, source }: TemplateVariableProps) {
...
+
+  // Once we computed value, we set it as the selected one, if it is available in the options
+  let selectedValue = value;
   if (!valueIsInOptions) {
-    selectValue = allowMultiple ? [] : '';
+    selectedValue = allowMultiple ? [] : '';
   }
 
+  if (ctx.definition.spec.name === 'instance') {
+    console.log('=> value', value);
```
Comment (Gladorme) [inline ui/dashboards/src/components/Variables/TemplateVariable.tsx:117]: 👀

**Review by sjcobb — APPROVED** [review]

File: ui/dashboards/src/components/Variables/TemplateVariable.tsx [hunk 2]
```diff
@@ -74,29 +82,68 @@ function ListVariable({ name, source }: TemplateVariableProps) {
...
+
+  // Update value when changed
   useEffect(() => {
-    const firstOption = finalOptions?.[0];
+    if (value) {
+      setVariableValue(name, value, source);
+    }
+  }, [setVariableValue, name, value, source]);
+
+  // Update loading when changed
```
Comment (sjcobb) [inline ui/dashboards/src/components/Variables/TemplateVariable.tsx:136]: Approved but I have a hunch that some of these hooks could be refactored and removed, didn’t get a chance to look into more specifics but newer React docs are emphasizing that useEffect can be unnecessary in many cases https://react.dev/learn/you-might-not-need-an-effect

Comment (celian-garcia) [reply ui/dashboards/src/components/Variables/TemplateVariable.tsx:136]: I fully agree to refactor it! Here I just encapsulate and test but for sure it doesn't make the code more readable, just a bit safer. Also in this particular case, it's tricky as we don´t want to update value too often or we have an infinite loop with the subsequent setVariableValue :smile:

Comment (celian-garcia) [reply ui/dashboards/src/components/Variables/TemplateVariable.tsx:136]: I can try to remove the useEffect for loading and options, but I'm sure that removing it for value would do some infinite loop...

**Review by Gladorme — APPROVED** [review]

**Pull request merged**